### PR TITLE
Redirect to `snark-verifier` and add more context on why this is rebuilt

### PR DESCRIPTION
**Pull request by han0110**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
 > ⚠️ This repo has NOT been audited and is NOT intended for a production environment yet.
 
-Solidity verifier generator for [`halo2`](http://github.com/privacy-scaling-explorations/halo2) proof with KZG polynomial commitment scheme on BN254
+Solidity verifier generator for [`halo2`](http://github.com/privacy-scaling-explorations/halo2) proof with KZG polynomial commitment scheme on BN254.
+
+For audited solidity verifier generator and proof aggregation toolkits, please refer to [`snark-verifier`](http://github.com/privacy-scaling-explorations/snark-verifier).
```
Comment (ChihChengLiang) [inline README.md]: The audited version is on Axiom's fork. Should we redirect there? 
```suggestion
For audited solidity verifier generator and proof aggregation toolkits, please refer to [`snark-verifier`](http://github.com/axiom-crypto/snark-verifier).
```

File: README.md [hunk 2]
```diff
@@ -30,16 +32,25 @@ let calldata = encode_calldata(vk_address, &proof, &instances);
...
 The [`Keccak256Transcript`](./src/transcript.rs#L19) behaves exactly same as the `EvmTranscript` in `snark-verifier`.
 
+## Why
+
+The existing solidity verifier generator in `snark-verifier` has a few problems:
+
+- Generator only receives low-level operation like add or mul, without hardwork to analyze these operations, it currently unrolled all the assembly codes, so it's easy to grow out of contract size limit even with a normal size circuit.
+- Complicated abstraction & APIs to consumers.
+
+So this repo is rebuilt from scratch, with focus also on codesize and readability, and surprisingly the gas cost is similar (even a little bit less) compared to the one generated by `snark-verifier`
```
Comment (ChihChengLiang) [inline README.md]: I applied some manual and GPT round editing to improve readability.
```suggestion
## Design Rationale

The current solidity verifier generator within `snark-verifier` faces a couple of issues:

- The generator receives only unoptimized, low-level operations, such as add or mul. As a result, it currently unrolls all assembly codes, making it susceptible to exceeding the contract size limit, even with a moderately sized circuit.
- The existing solution involves complex abstractions and APIs for consumers.

This repository is a ground-up rebuild, addressing these concerns while maintaining a focus on code size and readability. Remarkably, the gas cost is comparable, if not slightly lower, than the one generated by `snark-verifier`.
```

**Review by ChihChengLiang — APPROVED** [review]
LGTM. I added some suggestions

File: README.md (hunk 1 above)
Comment (han0110) [reply README.md]: Good point, thanks for catching this!

File: README.md (hunk 2 above)
Comment (han0110) [reply README.md]: Looks much better now!

**Pull request merged**